### PR TITLE
update typescript to generate type alias for implements IDictionary

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: specification/specs/common.ts [hunk 1]
```diff
@@ -103,8 +103,13 @@ class DateField {
   include_unmapped?: boolean;
 }
 
+/** Rather then documenting `object` this specifies places in the response that depend on the mapping and or request
+ * e.g the result of a script can be anything from value types to objects  */
 class UserDefinedValue {}
 
+/** Documents a place that can return ANY document from ANY index that can not be directly related back to a generic */
+class LazyDocument {}
```
Comment (delvedor) [inline specification/specs/common.ts:111]: Should we use this one for `MultiGet` as well?
Currently isn't a problem, as the `MultiGet` generic defaults to `unknown`.

Comment (Mpdreamz) [reply specification/specs/common.ts:111]: we should :)
https://github.com/elastic/elastic-client-generator/issues/71

**Comment by Mpdreamz** [comment]
This now uses the new TS magic to support the `IDictionary` trait that we translate to a recursive type alias.

Gets the search response failures down to `0` :smile_cat:

**Review by delvedor — CHANGES_REQUESTED** [review]
LGTM with a nit, can you add to the skip list the `AdditionalProperties` interface in the ts generator?

File: typescript-generator/index.js [hunk 2]
```diff
@@ -193,10 +197,43 @@ function isSpecialInterface (type) {
...
+  //find inherits on parent if current type has parent
+  //TODO fix spec so that inherits is no longer an array
+  if (Array.isArray(type.inherits)) return lookupBehaviorImplements(type.inherits[0], name)
+  return null;
+}
+
+function serializeAdditionalPropertiesType(type) {
+  const dictionaryOf = lookupBehaviorImplements(type, 'AdditionalProperties');
+  if (!dictionaryOf) throw new Error(`Unknown implements ${type.name.name}`)
+  //export type SingleBucketAggregate = { doc_count: double } & Record<string, Aggregate>
```
Comment (delvedor) [inline typescript-generator/index.js:212]: Leftover?

Comment (Mpdreamz) [reply typescript-generator/index.js]: Yep! removed

**Comment by Mpdreamz** [comment]
> > can you add to the skip list the AdditionalProperties interface in the ts generator? 
Done!

**Pull request merged**